### DISCUSSION TITLE
Update event cycle at runtime

### DISCUSSION THREAD

**Discussion by Abdjam** [COVESA/vsomeip#778] [Q&A]
Hello,

I am using the Vsomeip stack to provide a cyclic event and want the stack to handle the cyclic notifications. To achieve this, I’ve used the offer_event API and set the cycle value for my event. This works fine, but I need to change the event cycle at runtime. To do so, I call the stop_offer_event API and then the offer_event API again with the updated cycle value.

I've noticed that this approach works correctly when the new cycle value is smaller than the previous one. However, if the new cycle is larger, the stack continues sending events based on the old cycle.

My questions are:

Am I using the correct method to update the event cycle?
Can you explain the behavior I observed when using stop_offer_event followed by offer_event?

Thanks in advance

**Comment by duartenfonseca** [comment]
can you try to only change this on runtime and not set it on the config? this way maybe it would work. can you describe the use case where you would do this kinds of change?
